common/persistence/cassandra/cassandraPersistence.go

```diff
@@ -2284,54 +2292,51 @@ func (d *cassandraPersistence) LeaseTaskQueue(request *p.LeaseTaskQueueRequest)
 func (d *cassandraPersistence) UpdateTaskQueue(request *p.UpdateTaskQueueRequest) (*p.UpdateTaskQueueResponse, error) {
 	tli := *request.TaskQueueInfo
 	tli.LastUpdateTime = timestamp.TimeNowPtrUtc()
-	if tli.Kind == enumspb.TASK_QUEUE_KIND_STICKY { // if task_queue is sticky, then update with TTL
-		expiry := types.TimestampNow()
-		expiry.Seconds += int64(stickyTaskQueueTTL)
-
-		datablob, err := serialization.TaskQueueInfoToBlob(&tli)
-		if err != nil {
-			return nil, convertCommonErrors("UpdateTaskQueue", err)
-		}
+	datablob, err := serialization.TaskQueueInfoToBlob(&tli)
+	if err != nil {
+		return nil, convertCommonErrors("UpdateTaskQueue", err)
+	}
 
-		query := d.session.Query(templateUpdateTaskQueueQueryWithTTL,
+	var applied bool
+	previous := make(map[string]interface{})
+	if tli.Kind == enumspb.TASK_QUEUE_KIND_STICKY { // if task_queue is sticky, then update with TTL
+		batch := d.session.NewBatch(gocql.LoggedBatch)
```

Comment:
Any performance implications with this change? Before it was a simple insert, now it is a CAS insert + update?

Reply:
cass does not allow update setting ttl for primary key, hence the insert query.

we need to perform CAS for this query (see the normal task queue). the only additional cost is batch query, which only target at one row.

Reply:
Since we already run a background job to cleanup task queues on mysql backend, I think we should make this consistent across all databases and follow the same pattern for Cassandra.  Looks like this would simplify the logic significantly.

Reply:
> Since we already run a background job to cleanup task queues on mysql backend, I think we should make this consistent across all databases and follow the same pattern for Cassandra. Looks like this would simplify the logic significantly.

Short term fix vs long term solution

This PR aims to provide the short term fix, without too much perf impact

Reply:
Agreed.  Let's file a task for future improvement to eliminate TTL usage from cassandra persistence.

Reply:
#940